actionpack/lib/action_dispatch/middleware/cookies.rb

```diff
@@ -439,7 +451,23 @@ def write_cookie?(cookie)
           request.ssl? || !cookie[:secure] || always_write_cookie
         end
 
+        def default_options
+          options = request.cookies_default_options || {}
+
+          options.symbolize_keys!
+
+          options.transform_values { |v|
+            if v.is_a? Proc
```

Comment:
Perhaps this could open it up a bit for more advanced uses if need be?

```suggestion
            if v.respond_to?(:call)
```

Reply:
I wasn't sure which way to go, since the rails code base uses both. Your comment tipped the scale, thank you for the suggestion.